mypy.ini

```diff
@@ -4,6 +4,22 @@ mypy_path = ./app,./app/stubs
 plugins =
     mypy_django_plugin.main
 explicit_package_bases = True
+exclude = ^.*[/\\]test_.*
+
+# add some additional checks, but not quite strict
+strict = True
+disallow_untyped_calls = False
```

Comment:
What's the reason for this rule?

Reply:
> Disallows calling functions without type annotations from functions with type annotations.
Else we would need properly typed stubs for every not-typed library.

Reply:
I see that if we enable this, it only affects one place [in wsgi.py](https://github.com/geoadmin/service-control/blob/feat-pb-1056-tighten-mypy/app/wsgi.py#L78):
```py
StandaloneApplication(application, options).run()
```
because `Baseapplication.run()` is in the untyped package `gunicorn`.

There is the option to exclude untyped packages as described in [the docs for `disallow_untyped_calls` ](https://mypy.readthedocs.io/en/stable/config_file.html#confval-disallow_untyped_calls):
> If you want to selectively disable this check for all your code that imports some.library you should instead use [untyped_calls_exclude](https://mypy.readthedocs.io/en/stable/config_file.html#confval-untyped_calls_exclude), for example:
>
> ```toml
> [mypy]
> disallow_untyped_calls = True
> untyped_calls_exclude = some.library

So in our case:

```
untyped_calls_exclude = gunicorn
```

This is not going to resolve the case above because `StandaloneApplication` inherits from `gunicornBaseapplication`. Mypy is apparently too limited to take the inheritance hierarchy into account, see mypy issue ["--untyped-calls-exclude should consider inheritance"](https://github.com/python/mypy/issues/16651). So here, we would still need to a `# type: ignore`.

I would keep the rule because there must be good reasons that it's set as a default. But no strong opinion, to me it does not seem to be the most important one.

How do you see it given that there is `untyped_calls_exclude`?

Reply:
Nice, didn't know about this!